### PR TITLE
tests: bluetooth: tester: Make UART usage consistent

### PR DESCRIPTION
**Pull request by galak**
We use UART1 for UART pipe in a number of qemu tests, so lets do the
same thing for the bluetooth test and move the bluetooth uart to UART2.

Jira: ZEP-2412.

Signed-off-by: Savinay Dharmappa <savinay.dharmappa@intel.com>
Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

### CODE REVIEW

File: tests/bluetooth/tester/Makefile [hunk 1]
```diff
@@ -12,12 +12,12 @@ BOARD ?= arduino_101
...
 
-# UART for Bluetooth
-QEMU_EXTRA_FLAGS = -serial unix:/tmp/bt-server-bredr
-
 # UART for Tester
 QEMU_EXTRA_FLAGS += -serial pipe:/tmp/bt-stack-tester
 
+# UART for Bluetooth
+QEMU_EXTRA_FLAGS = -serial unix:/tmp/bt-server-bredr
+
```
Comment (nashif) [inline tests/bluetooth/tester/Makefile]: why is this being move here?

Comment (galak) [inline tests/bluetooth/tester/Makefile]: So we use the same serial port in qemu for the same purpose.  We utilize the 2nd qemu uart for UART_PIPE in a lot of other qemu tests.  So re-ordering things here so the PIPE in this test is 2nd UART and the BT uart is 3rd.

Comment (cvinayak) [inline tests/bluetooth/tester/Makefile]: What about missing "+=" when moved here?

**Review by cvinayak — DISMISSED** [review]

**Comment by SavinayDharmappa** [comment]
@galak  having CONFIG_UART_PIPE_ON_DEV_NAME in qemu.conf is irrelevant.  As we have following changes in drivers/console/Kconfig

if !HAS_DTS
config UART_PIPE_ON_DEV_NAME
        string "Device Name of UART Device for pipe UART"
        default "UART_0"
        depends on UART_PIPE
        help
        This option specifies the name of UART device to be used
        for pipe UART.
endif

**Comment by SavinayDharmappa** [comment]
@galak i tested you changes but bluetooth tester app doesn't work with it.

**Comment by Vudentz** [comment]
Ive been wondering myself if this would work, I guess the host tools would expect the ports in a certain order so changing it breaks the applications.

File: tests/bluetooth/tester/Makefile (hunk 1 above)
Comment (galak) [inline tests/bluetooth/tester/Makefile]: Fixed the +=, thanks

**Comment by galak** [comment]
@SavinayDharmappa the bluetooth tester doesn't work under what cfg?

**Comment by SavinayDharmappa** [comment]
@galak i pulled ur PR and built using following conf 
make BOARD=qemu_cortex_m3 CONF_FILE=qemu.conf

**Comment by Vudentz** [comment]
Btw, I don't think this has anything to do with consistency but priorities, what we should have is that UART_PIPE has higher priority than UART_BLUETOOTH or something else, we can't really always have the same port because each board has its own number of ports.

One problem we really need to solve is not have any build checks when 2 different drivers end up using the same port, but if the application set the ports on its own, which perhaps is what the tester should to avoid future conflicts, is to set the manually to whatever order the host application expects.

**Comment by galak** [comment]
@Vudentz what do you mean by priority?  Is there some conflict with bt & uart-pipe?

**Comment by galak** [comment]
@SavinayDharmappa the README doesn't talk at all about how to setup /tmp/bt-stack-tester.

**Comment by galak** [comment]
@SavinayDharmappa any update on my question, and how I can try and reproduce this.

**Comment by SavinayDharmappa** [comment]
@galak  we have not explictly setup /tmp/bt-stack-tester. Just followed the README

**Comment by rmstoi** [comment]
/tmp/bt-stack-tester is created by the auto-pts, the Bluetooth PTS testing automation tool.

With this pull request Bluetooth tester does not work. How I test:

Build the tester:

`make pristine && make BOARD=qemu_cortex_m3 CONF_FILE=qemu.conf`

Then test the tester with auto-pts btpclient (you could also run a test caset if you want).

Successful test is:

```
[btpclient]$ start-zephyr /zephyr/tests/bluetooth/tester/outdir/qemu_cortex_m3/zephyr.elf
created fd /tmp/bt-stack-tester, listening...
btp server connected successfully

[btpclient]$ core gap_reg
Received header(svc_id=0, op=0x80, ctrl_index=255, data_len=0)
Received data (hex): 
Received data (ascii): ('',)
Invalid opcode 0x80 in the response, expected 0x03!
Expected svc_id=0, op=0x03

[btpclient]$ core gap_reg
Received header(svc_id=0, op=0x03, ctrl_index=255, data_len=0)
Received data (hex): 
Received data (ascii): ('',)
OK
```
First time running "core gap_reg" received "IUT ready event" from tester. The second time, gap is registered successfuly.

I tested this PR with tester app and it does not work. auto-pts blocks while waiting for the "IUT ready event" from the tester.

With PR #914 tester works.

**Comment by galak** [comment]
@rmstoi So if I do the the make line I get:

To exit from QEMU enter: 'CTRL+a, x'
[QEMU] CPU: cortex-m3
qemu-system-arm: -serial pipe:/tmp/bt-stack-tester: Could not open '/tmp/bt-stack-tester': No such file or directory
qemu-system-arm: -serial pipe:/tmp/bt-stack-tester: could not connect serial device to character backend 'pipe:/tmp/bt-stack-tester'
/home/galak/git/gh/zephyr/scripts/Makefile.qemu:26: recipe for target 'run' failed
make[2]: *** [run] Error 1
make[2]: Leaving directory '/home/galak/git/gh/zephyr/tests/bluetooth/tester/outdir/qemu_cortex_m3'
Makefile:178: recipe for target 'sub-make' failed
make[1]: *** [sub-make] Error 2
make[1]: Leaving directory '/home/galak/git/gh/zephyr'
/home/galak/git/gh/zephyr/Makefile.inc:97: recipe for target 'run' failed
make: *** [run] Error 2

**Comment by galak** [comment]
@rmstoi - What is 'Bluetooth PTS testing automation tool'?

**Comment by galak** [comment]
@rmstoi how do things fail if you use PR #926?

**Comment by rmstoi** [comment]
@galak 
With PR #926 "IUT ready event"  from Bluetooth tester is not received over UART. So communication with tester is not working.

**Comment by rmstoi** [comment]
I would suggest to revert c1e5cb to get the tester working. As mentioned here:

https://jira.zephyrproject.org/browse/ZEP-2412

**Comment by galak** [comment]
I reverted the commit, and still get:

qemu-system-arm: -serial pipe:/tmp/bt-stack-tester: Could not open '/tmp/bt-stack-tester': No such file or directory
qemu-system-arm: -serial pipe:/tmp/bt-stack-tester: could not connect serial device to character backend 'pipe:/tmp/bt-stack-tester'
/home/galak/git/gh/zephyr/scripts/Makefile.qemu:26: recipe for target 'run' failed


I think this is because I'm missing some aspect of what 'Bluetooth PTS testing automation tool' is.

**Comment by rmstoi** [comment]
We could simulate auto-pts a bit.

First terminal:

`socat -x -v UNIX-LISTEN:/tmp/bt-stack-tester,reuseaddr,fork -`

Second terminal:

`make pristine && make BOARD=qemu_cortex_m3 CONF_FILE=qemu.conf`

`/opt/zephyr-sdk/sysroots/x86_64-pokysdk-linux/usr/bin/qemu-system-arm -cpu cortex-m3 -machine lm3s6965evb -nographic -vga none -pidfile qemu.pid -serial mon:stdio -serial unix:/tmp/bt-server-bredr -serial unix:/tmp/bt-stack-tester -kernel outdir/qemu_cortex_m3/zephyr.elf`

You could also use "make run", but need to modify "pipe" to "unix" in the Makefile

With PR #926 nothing is printed in socat, whereas with PR #914: "IUT ready event" is received:

```
> 2017/08/24 11:12:31.036662  length=5 from=0 to=4
 00 80 ff 00 00                                   .....
```

**Comment by rmstoi** [comment]
I tested this PR again and it works.

My previous QEMU command was old,  because of UART changes, /tmp/bt-stack-tester should come before /tmp/bt-server-bredr. This old command was in the test scripts, after fixing that, test cases pass.

**Comment by galak** [comment]
@rmstoi which test scripts are you speaking of?  one's in the Zephyr code base?

**Comment by rmstoi** [comment]
@galak those are some private test scipts

**Pull request merged**